addons/Popochiu/Engine/Objects/Room/PopochiuRoom.gd

```diff
@@ -247,7 +273,12 @@ func _update_navigation_path(
 ):
 	# TODO: Use a Dictionary so more than one character can move around at the
 	# same time. Or maybe each character should handle its own movement? (;￢＿￢)
-	_path = _nav_path.get_simple_path(start_position, end_position, true)
+	if character.ignore_walkable_areas:
+		# if the character can glide between WAs, just move over a straight line
+		_path = PoolVector2Array([start_position, end_position])
+	else:
+		# if the character is forced into WAs, delegate pathfinding to the active WA
+		_path = _nav_path.get_simple_path(start_position, end_position, true)
```

Comment:
I really like this!

Reply:
Aw, thanks. It's pretty simple, I thought about a more convoluted pathfinding but if you have a specific necessity in game, the best option is you script stuff so the character moves to specific points before leaving a WA.
This gives you more control, less guesswork and simpler code at engine level.